src/debugger.ts

```diff
@@ -2,32 +2,68 @@ import * as vscode from 'vscode';
 import * as path from 'node:path';
 import { globalContext } from './extension';
 import { MirrordConfigManager } from './config';
-import { LAST_TARGET_KEY, MirrordAPI, mirrordFailure } from './api';
+import { LAST_TARGET_KEY, MirrordAPI, mirrordFailure, MirrordExecution } from './api';
 import { updateTelemetries } from './versionCheck';
 import { getLocalMirrordBinary, getMirrordBinary } from './binaryManager';
+import { platform } from 'node:os';
 
-/// Get the name of the field that holds the exectuable in a debug configuration of the given type.
-function getExecutableFieldName(config: vscode.DebugConfiguration): keyof vscode.DebugConfiguration {
+const DYLD_ENV_VAR_NAME = "DYLD_INSERT_LIBRARIES";
+
+/// Get the name of the field that holds the exectuable in a debug configuration of the given type,
+/// and the executable. Returning the field name for replacing the value with the patched path later.
+/// Also returning the executable because in some configuration types there is some extra logic to
+/// be done for retrieving the executable out of its field (see the `node-terminal` case).
+function getFieldAndExecutable(config: vscode.DebugConfiguration): [keyof vscode.DebugConfiguration, string | null] {
 	switch (config.type) {
 		case "pwa-node":
 		case "node": {
-			return "runtimeExecutable";
+			return ["runtimeExecutable", config["runtimeExecutable"]];
+		}
```

Comment:
can you add some docs here, where are we getting these types from? also make sure if this change maps directly to pwa-node

Reply:
> can you add some docs here, where are we getting these types from?

Added link to vscode's docs of the node launch configuration configuration.

> also make sure if this change maps directly to pwa-node

What do you mean?